hlapipeline/alignimages.py

```diff
@@ -373,7 +428,20 @@ def perform_align(input_list, archive=False, clobber=False, update_hdr_wcs=False
                 filteredTable[index]['rms_dec'] = item.meta['tweakwcs_info']['RMS_DEC'].value
                 filteredTable[index]['fit_rms'] = item.meta['tweakwcs_info']['FIT_RMS']
                 filteredTable[index]['total_rms'] = item.meta['tweakwcs_info']['TOTAL_RMS']
-                #filteredTable.pprint(max_width=-1)
+                # populate filteredTable fields "status", "compromised" and
+                # "processMsg" with fitStatusDict fields "valid", "compromised"
+                # and "reason".
+                explicitDictKey ="{},{}".format(item.meta['name'], item.meta['chip'])
+                if fitStatusDict[explicitDictKey]['valid'] == True:
+                    filteredTable[index]['status'] = 0
+                else:
+                    filteredTable[index]['status'] = 1
+                if fitStatusDict[explicitDictKey]['compromised'] == False:
+                    filteredTable['compromised'] = 0
+                else:
+                    filteredTable['compromised'] = 1
+                if fitStatusDict[explicitDictKey]['reason'] != "":
+                    filteredTable[index]['processMsg'] = fitStatusDict[explicitDictKey]['reason']
```

Comment:
Processing IC8OA5020, this (expected) failure case did not update the filteredTable with the failure reason:

```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
nmatchesCheck: True radialOffsetCheck: True largeRmsCheck: False, consistencyCheck: True
{'valid': False, 'max_rms': 1188.3599062543028, 'num_matches': 40, 'compromised': False, 'reason': 'RMS too large (>150 mas)!'}
{'valid': False, 'max_rms': 1188.3599062543028, 'num_matches': 39, 'compromised': False, 'reason': 'RMS too large (>150 mas)!'}
{'valid': False, 'max_rms': 1188.3599062543028, 'num_matches': 42, 'compromised': False, 'reason': 'RMS too large (>150 mas)!'}
{'valid': False, 'max_rms': 1188.3599062543028, 'num_matches': 43, 'compromised': False, 'reason': 'RMS too large (>150 mas)!'}
Total fit RMS value = 1188.3599062543028 mas greater than the maximum threshold value 10.
The fit solution for some or all of the images is not valid.
Try again with the next catalog
FIT SOLUTION REJECTED
Processing time of [STEP 5b]: 0.355544 sec
-------------------- STEP 6: Collect up information and populate the filtered table --------------------
The fitting process was unsuccessful with a best fit total rms of -99999.0 mas
Processing time of [STEP 6]: 1.5e-05 sec
-------------------- STEP 7: Update image headers with new WCS information --------------------

 STEP SKIPPED
Processing time of [STEP 7]: 1.3e-05 sec
TOTAL Processing time of 10.687669 sec
{}

In [26]: align_table                                                                                                                
Out[26]: 
<Table length=4>
    imageName      instrument detector  filter aperture obstype ... completed fit_rms total_rms datasetKey status headerletFile
     bytes20        bytes20   bytes20  bytes20 bytes20  bytes20 ...    int8   float64  float64    int64    int32     bytes30   
------------------ ---------- -------- ------- -------- ------- ... --------- ------- --------- ---------- ------ -------------
ic8oa5tcq_flt.fits       WFC3       IR   F125W   IR-FIX IMAGING ...         0    -1.0      -1.0         -1      1          None
ic8oa5teq_flt.fits       WFC3       IR   F125W   IR-FIX IMAGING ...         0    -1.0      -1.0         -1      1          None
ic8oa5tpq_flt.fits       WFC3       IR   F125W   IR-FIX IMAGING ...         0    -1.0      -1.0         -1      1          None
ic8oa5ttq_flt.fits       WFC3       IR   F125W   IR-FIX IMAGING ...         0    -1.0      -1.0         -1      1          None

In [28]: align_table['processMsg']                                                                                                  
Out[28]: 
<Column name='processMsg' dtype='bytes20' length=4>
None
None
None
None
```
The solution included in this traceback section here indicates that a fit was performed successfully, but it was obviously bad (RMS > 500mas for each image).  However, It would be useful in cases like this for the filteredTable to get some information about the cause of the failure, **if at all possible**.  In this case, passing along the `reason` from the `fitStatusDict` as the `processMsg`.

Reply:
Not quite done yet, but issue mostly fixed by commit [11ce3d0](https://github.com/spacetelescope/hlapipeline/pull/73/commits/11ce3d012ac4665035abd7091aae6636a18cfd13)

Reply:
@stsci-hack and @mdlpstsci, I increased the length of processMsg from 20 to 30 characters to accommodate more descriptive messages in commits  [a0f0676(https://github.com/spacetelescope/hlapipeline/pull/73/commits/a0f0676f93783b4da5e13619da76aa8762a59845) and [8c1b293](https://github.com/spacetelescope/hlapipeline/pull/73/commits/8c1b293714949e022d059b58ec71fc7fc0967ba5)

Reply:
Issue Resolved